x/btcstaking/keeper/finality_providers.go

```diff
@@ -57,10 +62,13 @@ func (k Keeper) AddFinalityProvider(goCtx context.Context, msg *types.MsgCreateF
 		CommissionInfo: commissionInfo,
 	}
 
-	if bsnID == ctx.ChainID() {
-		k.setFinalityProvider(ctx, &fp)
-	} else {
-		if err := k.SetConsumerFinalityProvider(ctx, &fp, bsnID); err != nil {
+	k.setFinalityProvider(ctx, &fp)
+	k.bsnIndexFinalityProvider(ctx, &fp)
+
+	// Create BTC Staking Consumer Event for the new finality provider
+	// TODO: examine whether these events are really needed
+	if !fp.SecuresBabylonGenesis(ctx) {
```

Comment:
I'm not sure if it's a good idea to refer to "Genesis" in the codebase. It refers to Babylon Genesis mainnet but the codebase is generic to any Babylon network, e.g., testnets

Reply:
Babylon Genesis is the network operated by the node in this repository. Babylon is the protocol. For example, our testnet is the Babylon Genesis testnet, not the Babylon testnet. In the long run, we should aim to start using "Babylon Genesis" more to refer to the chain, as just "Babylon" is not correct terminology.

Nonetheless, if you see my comment on the `SecuresBabylonGenesis` function, it is a temporary one that should be replaced by a better heuristic for identifying whether a finality provider secures genesis or not.